editor/components/skip-to-selected-block/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getBlockWrapperDOMNode } from '../../utils/dom';
+
+const SkipToSelectedBlock = ( { selectedBlock } ) => {
+	if ( ! selectedBlock ) {
+		return null;
+	}
+
+	const { uid } = selectedBlock;
+
+	const onClick = () => {
+		const selectedBlockElement = getBlockWrapperDOMNode( uid );
+		selectedBlockElement.focus();
+	};
+
+	return (
+		selectedBlock && uid &&
+		<button type="button" className="button skip-to-selected-block" onClick={ onClick }>
+			{ __( 'Skip to the selected block' ) }
+		</button>
+	);
+};
+
+export default withSelect( ( select ) => {
+	return {
+		selectedBlock: select( 'core/editor' ).getSelectedBlock(),
```

Comment:
`getSelectedBlock` can be a computationally-intensive selector. I wish we had a simpler way to get access to the direct UID value of the singular selected block. We do have `getBlockSelectionStart`, but this also returns a value for a multi-selection. Which actually raises another point: do we care if it's a singular block selection? Seems like we'd want to return to the first UID of the set of selected blocks in either case.